src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -279,7 +299,7 @@ private void PreProcessOnStatement(ParseNode node)
             PushTriggerRemoveName(triggerRemoveVarName);
             AddOpcode(new OpcodePush(triggerRemoveVarName));
             AddOpcode(new OpcodePush(true));
-            AddOpcode(new OpcodeStore());
+            AddOpcode(new OpcodeStoreGlobal());
```

Comment:
The "remove-trigger-number....." var name can be always global because the number part of the name is the unique hashcode of the instance of the trigger body we're in.  In fact it's better to force it to be global because it shouldn't live in whatever runtime location happened to get interrupted by this trigger.